R/summary_funs.R

```diff
@@ -202,198 +209,223 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
-    if (!is.null(lppd.bs)) {
-      # Compute the performance statistics using only those observations for
-      # which both `lppd` and `lppd.bs` are not `NA`:
-      lppd[is.na(lppd.bs)] <- NA
-      lppd.bs[is.na(lppd)] <- NA
-      n_notna.bs <- sum(!is.na(lppd.bs))
-    }
-    n_notna <- sum(!is.na(lppd))
-    n <- length(lppd)
-  } else {
-    hasNA_y <- is.na(y_wobs_test$y_prop %||% y_wobs_test$y)
-    if (!is.null(mu.bs)) {
-      # Compute the performance statistics using only those observations for
-      # which both `mu` and `mu.bs` are not `NA`:
-      mu[is.na(mu.bs)] <- NA
-      mu.bs[is.na(mu)] <- NA
-      n_notna.bs <- sum(!is.na(mu.bs) & !hasNA_y)
-    }
-    n_notna <- sum(!is.na(mu) & !hasNA_y)
-    n <- length(mu)
-  }
-  if (!is.null(n_notna.bs) && getOption("projpred.additional_checks", FALSE)) {
-    stopifnot(n_notna == n_notna.bs)
-  }
-  if (n_notna == 0) {
+get_stat <- function(summaries, summaries_baseline = NULL,
+                     summaries_fast = NULL,
+                     y_wobs_test, stat, alpha = 0.1, ...) {
+  mu <- summaries$mu
+  lppd <- summaries$lppd
+  loo_inds <- which(!is.na(lppd))
+  n <- length(lppd)
+  n_loo <- length(loo_inds)
+  ## is this needed anymore?
+  ## n_notna.bs <- NULL
+  ## if (!is.null(summaries_fast)) {
+  ##   # Compute the performance statistics using only those observations for
+  ##   # which fast summaries are not NA
+  ##   if (stat %in% c("elpd", "mlpd", "gmpd")) {
+  ##     lppd[is.na(lppd_baseline)] <- NA
+  ##     n_notna.bs <- sum(!is.na(lppd_baseline))
+  ##   }
+  ##   n_notna <- sum(!is.na(lppd))
+  ##   n <- length(lppd)
+  ## } else {
+  ##   hasNA_y <- is.na(y_wobs_test$y_prop %||% y_wobs_test$y)
+  ##   if (!is.null(mu_baseline)) {
+  ##     # Compute the performance statistics using only those observations for
+  ##     # which both `mu` and `mu_baseline` are not `NA`:
+  ##     mu[is.na(mu_baseline)] <- NA
+  ##     mu_baseline[is.na(mu)] <- NA
+  ##     n_notna.bs <- sum(!is.na(mu_baseline) & !hasNA_y)
+  ##   }
+  ##   n_notna <- sum(!is.na(mu) & !hasNA_y)
+  ##   n <- length(mu)
+  ## }
+  ## if (!is.null(n_notna.bs) && getOption("projpred.additional_checks", FALSE)) {
+  ##   stopifnot(n_notna == n_notna.bs)
+  ## }
+  if (n_loo == 0) {
     return(list(value = NA, se = NA, lq = NA, uq = NA))
   }
-
-  if (is.null(wcv)) {
-    ## set default CV fold weights if not given
-    wcv <- rep(1, n)
-  }
-  ## ensure the CV fold weights sum to n_notna
-  wcv <- n_notna * wcv / sum(wcv)
-
   alpha_half <- alpha / 2
   one_minus_alpha_half <- 1 - alpha_half
 
   if (stat %in% c("elpd", "mlpd", "gmpd")) {
-    if (!is.null(lppd.bs)) {
-      value <- sum((lppd - lppd.bs) * wcv, na.rm = TRUE)
-      value.se <- weighted.sd(lppd - lppd.bs, wcv, na.rm = TRUE) *
-        sqrt(n_notna)
+    if (is.null(summaries_baseline)) {
+      lppd_baseline = 0
+    } else {
+      lppd_baseline = summaries_baseline$lppd
+    }
+    if (!is.null(summaries_fast) && sum(n_loo<n)) {
```

Comment:
Why `sum()` here? I think `n_loo` and `n` should both be scalars, right?

Reply:
No idea where it came from, but all fixed